### PR TITLE
 feat: add address form

### PR DESCRIPTION
**Pull request by alexj105**
The type of this PR is: feature

This PR resolves [PURCHASE-2760]

### Description
without tests at the moment
add new address form + as default option

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a CHANGELOG.yml entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[PURCHASE-2760]: https://artsyproduct.atlassian.net/browse/PURCHASE-2760

https://user-images.githubusercontent.com/37010996/125648691-7f43a976-9a35-4bf3-ad78-8b7ba8e4e551.mov

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fb308354c73cd312794ba9fddef084a8444f92cb
</p>

File: CHANGELOG.yml [hunk 1]
```diff
@@ -4,6 +4,34 @@ upcoming:
...
+    - Fix visual bug on auction result list item on artist insights screen - ole
+    - Fix auction result home rail loading spinner issue - ole
+    - Fix share button on BNMO Flow - ole, mounir, sam, george
+    - Annotate home screen queries with optionalField directive - ole
+    - Fix collection home rail loading spinner issue - ole
+    - Refactor category screen header and filters to match other artwork grids - dzmitry tratsiak
+    - Use infinite scroll in featured fairs screen - katsiaryna alshannikava, ole
+    - Fix overlapping of Gallery name and time remaining for Vieweing rooms - katsiaryna alshannikava
+    - Fix auction results for you description text scroll issue - ole
+    - Fix facebook login failing after signing up for a new account - stanislau hanchar
```
Comment (jo-rs) [inline CHANGELOG.yml:34]: How come all these are marked as changes? Did something go wrong on a merge? 🤔

File: Podfile.lock [hunk 2]
```diff
@@ -778,7 +778,14 @@ DEPENDENCIES:
...
 SPEC REPOS:
-  https://cdn.cocoapods.org/:
+  https://github.com/artsy/Specs.git:
+    - Aerodramus
+    - "Artsy+UIColors"
+    - "Artsy+UIFonts"
+    - "Artsy+UILabels"
+    - Artsy-UIButtons
+    - Extraction
+  trunk:
```
Comment (jo-rs) [inline Podfile.lock:788]: Are these changes intentional?

File: Podfile.lock [hunk 3]
```diff
@@ -848,13 +855,6 @@ SPEC REPOS:
     - "UIView+BooleanAnimations"
     - "XCTest+OHHTTPStubSuiteCleanUp"
     - YogaKit
-  https://github.com/artsy/Specs.git:
-    - Aerodramus
-    - "Artsy+UIColors"
-    - "Artsy+UIFonts"
-    - "Artsy+UILabels"
-    - Artsy-UIButtons
-    - Extraction
```
Comment (jo-rs) [inline Podfile.lock:857]: Are these changes intentional?

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx [hunk 4]
```diff
@@ -97,7 +98,13 @@ const SavedAddresses: React.FC<{ me: SavedAddresses_me; relay: RelayRefetchProp
...
-            <Button block variant="primaryBlack" width={100}>
+            <Button
+              block
+              borderRadius={50}
+              variant="primaryBlack"
+              width={100}
+              onPress={() => navigate("/my-profile/saved-addresses/new-address")}
+            >
               Add New Address
             </Button>
```
Comment (jo-rs) [inline src/lib/Scenes/SavedAddresses/SavedAddresses.tsx:109]: We should probably extract this button since we also want it on the list view, no?

File: src/lib/Scenes/SavedAddresses/SavedAddressesNewForm.tsx [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+            <PlaceholderText height={10} width={45 + Math.random() * 100} />
+            <PlaceholderText height={30} width={150 + Math.random() * 100} />
+            <PlaceholderText height={10} width={45 + Math.random() * 100} />
+            <PlaceholderText height={30} width={150 + Math.random() * 100} />
+          </Flex>
+        ))}
+      </Flex>
+    </PageWithSimpleHeader>
+  )
+}
```
Comment (jo-rs) [inline src/lib/Scenes/SavedAddresses/SavedAddressesNewForm.tsx:200]: I'm a bit confused about what this is about 🤔

File: src/lib/Scenes/SavedAddresses/addNewAddress.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import {
```
Comment (jo-rs) [inline src/lib/Scenes/SavedAddresses/addNewAddress.ts:1]: I'd say let's move the two mutations to a `/mutations` folder.

**Review by jo-rs — COMMENTED** [review]
It looks great 💯  
I'm just a bit confused by a couple of things.

File: src/lib/Scenes/SavedAddresses/__tests__/SavedAddresses-tests.tsx [hunk 7]
```diff
@@ -109,4 +110,20 @@ describe(SavedAddressesQueryRenderer, () => {
     expect(text).toContain("Edit")
     expect(text).toContain("Delete")
   })
+  it("renders SaveAddressNewForm component", () => {
```
Comment (jo-rs) [inline src/lib/Scenes/SavedAddresses/__tests__/SavedAddresses-tests.tsx:113]: Any other tests that can be added?

File: Podfile.lock (hunk 2 above)
Comment (alexj105) [reply Podfile.lock:788]: > Are these changes intentional?

I get it after pod install, so I think yes

File: src/lib/Scenes/SavedAddresses/__tests__/SavedAddresses-tests.tsx (hunk 7 above)
Comment (alexj105) [reply src/lib/Scenes/SavedAddresses/__tests__/SavedAddresses-tests.tsx:55]: this test is for testing navigation after add address button click. SaveAddressNewForm is tested in a separate file, I renamed this case. SavedAddresses component isn't the context of my pr. Anyway, if you have some tests cases for this component I would be glad to add it  ;)

File: src/lib/Scenes/MyAccount/Components/MyAccountFieldEditScreen.tsx [hunk 8]
```diff
@@ -77,18 +80,20 @@ export const MyAccountFieldEditScreen = React.forwardRef<
...
           </TouchableOpacity>
         }
         title={title}
         right={
-          <TouchableOpacity disabled={!canSave} onPress={handleSave}>
-            <Sans size="4" opacity={!canSave ? 0.3 : 1}>
-              Save
-            </Sans>
-          </TouchableOpacity>
+          !(isSaveButtonVisible === false) && (
```
Comment (jo-rs) [inline src/lib/Scenes/MyAccount/Components/MyAccountFieldEditScreen.tsx:90]: This can probably be simplified, no? 😬

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx [hunk 9]
```diff
@@ -24,14 +26,33 @@ const Card = styled(Flex)`
...
+      })
+    }
+    __triggerRefresh = triggerRefresh
+    return () => {
+      if (__triggerRefresh === triggerRefresh) {
+        __triggerRefresh = null
+      }
+    }
+  }, [])
+
```
Comment (jo-rs) [inline src/lib/Scenes/SavedAddresses/SavedAddresses.tsx:50]: I think we can take out this refresh logic for now. There's an easier way of doing it, and there's a ticket for it 😊

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx [hunk 10]
```diff
@@ -105,9 +126,10 @@ const SavedAddresses: React.FC<{ me: SavedAddresses_me; relay: RelayRefetchProp
             <Text variant="caption" textAlign="center" mb={3}>
               Please add an address for a faster checkout experience in the future.
             </Text>
-            <Button block variant="primaryBlack" width={100}>
-              Add New Address
-            </Button>
+            <AddAddressButton
+              handleOnPress={() => navigate("/my-profile/saved-addresses/new-address")}
+              title="Add New Address"
+            />
```
Comment (jo-rs) [inline src/lib/Scenes/SavedAddresses/SavedAddresses.tsx:132]: Is this button currently showing up both for the empty screen and not empty one?

**Review by jo-rs — COMMENTED** [review]
Just a couple of small changes, and a question, and I think we're ready to go 🚀 
Let's touch base in the morning if anything is not clear.

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx (hunk 10 above)
Comment (alexj105) [reply src/lib/Scenes/SavedAddresses/SavedAddresses.tsx:132]: just when our list is empty (ListEmptyComponent in Flatlist)
Screen Shot 2021-07-16 at 9 45 42 AM

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx (hunk 9 above)
Comment (alexj105) [reply src/lib/Scenes/SavedAddresses/SavedAddresses.tsx:50]: will be glad to see another solution. dropped it 👍

File: src/lib/Scenes/MyAccount/Components/MyAccountFieldEditScreen.tsx (hunk 8 above)
Comment (alexj105) [reply src/lib/Scenes/MyAccount/Components/MyAccountFieldEditScreen.tsx:90]: yeap, got a little carried away ;)

**Review by jo-rs — APPROVED** [review]
🚀
